src/main/java/org/hiero/sketch/table/api/IMembershipSet.java

```diff
@@ -15,6 +15,26 @@
      * @return Total number of elements in this membership map.
      */
     int getSize();
+
+    /**
+     * @return If exact==true returns the total umber of elements in this membership map. If exact==false may
+     * return an approximation of that number. An approximation may be much quicker to compute so if an approximation
+     * is sufficient it is recommended to set exact==false.
+     */
+    int getSize(boolean exact);
+    /**
+     * @return an ImembershipSet containing k samples from the membership map. The samples are made
+     * with replacement so may contain less than k distinct values. There is no guarantee that two subsequent samples
+     * return the same sample set.
+     */
+    IMembershipSet sample(int k);
+
+    /**
+     * @return an ImembershipSet containing k samples from the membership map. The samples are made
+     * with replacement so may contain less than k distinct values. The pseudo-random generated
+     * is seeded with parameter seed, so subsequent calls with the same seed are guaranteed to return the same sample.
+     */
+    IMembershipSet sample(int k, long seed);
```

Comment:
Please add here some binary set operations: intersection, union, difference and symmetric difference at least.